### ISSUE TITLE
Package "is-extglob" not being linked

### ISSUE THREAD

**Issue by nickpape-msft** [pnpm/pnpm#1305]
### Code to reproduce the issue:

I wasn't able to get a minimalist repro, and the repo that it does occur in contains a bunch of references to private packages.

Our shinkwrap contains:
```yaml
  /is-glob/3.1.0:
    dependencies:
      is-extglob: 2.1.1
    dev: false
    engines:
      node: '>=0.10.0'
    resolution:
      integrity: sha1-e6WuJCF4BKxwcHuWkiVnSGzD6Eo=
```

### Actual behavior: "is-extglob" is not linked.

Oddly, it does appear in the store:

![image](https://user-images.githubusercontent.com/17131271/43165570-142ba79c-8f49-11e8-9471-df79e93698e4.png)

And it appears in the node_modules:

![image](https://user-images.githubusercontent.com/17131271/43165617-361a1370-8f49-11e8-94b9-a58b494df6b2.png)

But it is not linked by "is-glob"; 

![image](https://user-images.githubusercontent.com/17131271/43165683-638abad0-8f49-11e8-8ab9-dbbd78e2b1f8.png)

It's deterministic and occurs on a clean install.

The rush/pnpm log is here: 
[InstallLog.txt](https://github.com/pnpm/pnpm/files/2225385/InstallLog.txt)

### Additional information:

 - `node -v` prints: 8.9.4
 - Windows, OS X, or Linux?: Windows Server 2008 R2

**Comment by nickpape-msft** [comment]
@pgonzal @iclanton @zkochan @lynamemi FYI

**Comment by pgonzal** [comment]
Let us know if there's any other info that you need regarding this issue.  Thanks!

**Comment by zkochan** [comment]
I did look into the logs. I could not figure out what the issue is.
If you email me the `shrinkwrap.yaml` file, maybe it'll help.
If this is consistently reproducible, maybe you could debug it yourself and find the cause (also, does it also happen on Linux or OSX?)

This is the piece of code that links to node_modules:

https://github.com/pnpm/pnpm/blob/0b71b3d1be753c67994056f7535409397043860e/packages/supi/src/link/index.ts#L441-L472

so maybe you could debug it or we could add more logs but I don't know yet what logs would help to locate this issue.

Making a small repro would also help a lot. You could try to remove dependencies one-by-one, till the issue goes aways and try to find the set of dependencies that make the issue. I know this is time consuming but it would make it a lot easier to find the issue

**Comment by zkochan** [comment]
@pgonzal I have a guess. In the logs I see that the wanted `shrinkwrap.yaml` does not match the current one. The only way it may happen in your case is if your run `pnpm install` with `--no-optional`.

Can you try if this is reproducible with `--no-optional`?

If it is not reproducible when optional deps are not skipped then send me the `shrinkwrap.yaml` and `node_modules/.shrinkwrap.yaml` files.

**Comment by pgonzal** [comment]
Sure, I'll follow up with @nickpape-msft when he gets in today.  I believe he still has a repro.

**Comment by pgonzal** [comment]
Maybe there is a `console.log` statement that Nick could inject into **link/index.js** file that would help you understand why it's not symlinking that file?

**Comment by pgonzal** [comment]
@ThomasMichon FYI

**Comment by zkochan** [comment]
please retry with pnpm v2.14.1 and attach the logs

I added additional logging

**Comment by zkochan** [comment]
Excuse me, the publish didn't go well. Use v2.14.2

**Comment by lynamemi** [comment]
I tried the build again (still failing) but it looks like we're on version 2.13.5 anyway. @nickpape-msft do you know if there are plans to bump the version to the one noted above?

**Comment by pgonzal** [comment]
> do you know if there are plans to bump the version to the one noted above?

 @lynamemi which repo are you referring to?

**Comment by lynamemi** [comment]
@pgonzal odsp-common

**Comment by pgonzal** [comment]
That repo is self-service.  Feel free to create a PR bumping the `"pnpmVersion"` field in `rush.json` file.

**Comment by zkochan** [comment]
I located the issue. I don't know what causes it yet but I know the line code that causes it.

The issue is caused because pnpm thinks for some reason that `is-extglob` is not installable, so it is not linked.

https://github.com/pnpm/pnpm/blob/b685487358eb2465bf62e3a988b4484115992171/packages/supi/src/install/link.ts#L504

**Comment by zkochan** [comment]
fixed in v2.16.3
